python/ray/serve/tests/test_custom_metrics.py

```diff
@@ -0,0 +1,117 @@
+import sys
+from typing import Any, Dict
+
+import pytest
+
+import ray
+from ray import serve
+from ray._common.test_utils import SignalActor, wait_for_condition
+from ray.serve._private.common import DeploymentID
+from ray.serve._private.test_utils import (
+    send_signal_on_cancellation,
+)
+
+
+def get_autoscaling_metrics_from_controller(
+    client, deployment_id: DeploymentID
+) -> Dict[str, float]:
+    """Get autoscaling metrics from the controller for testing."""
+    ref = client._controller._dump_all_autoscaling_metrics_for_testing.remote()
+    metrics = ray.get(ref)
+    return metrics.get(deployment_id, {})
+
+
+class TestCustomServeMetrics:
+    """Check that redeploying a deployment doesn't reset its start time."""
+
+    def test_custom_serve_metrics(self, serve_instance):
+        @serve.deployment(
+            autoscaling_config={
+                "min_replicas": 1,
+                "max_replicas": 5,
+                "target_num_ongoing_requests_per_replica": 2,
```

Comment:
`target_num_ongoing_requests_per_replica` is not a valid `autoscaling_config`, instead use `target_ongoing_requests`. Also you should leave this config out since its not needed for custom autoscaling metrics ?